### PR TITLE
[IMP] point_of_sale: Add the possibility to the IoT Box to perform a self-flashing

### PR DESCRIPTION
**Pull request by qle-odoo**
To be able to flash the partition on which the IoT Box is executed,
    we must create a third partition and install an OS there to be able
    to boot on it and flash the partition of the IoT Box

    - Creation of the third partition
    - Raspbian download and checksum verification
    - Installation of Raspbian on the third partition
    - Download the IoT image on the Raspbian partition and check the checksum
    - Reboot on the Raspbian partition
    - Installation of the IoT image on the second partition
    - Reboot on the IoT partition
    - Cleaning and deleting the third partition

    Task: 2161955

### CODE REVIEW

File: addons/hw_posbox_homepage/views/upgrade_page.html [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+            $('#flash').click(async function() {
+                if (confirm('Alert message to flash IoT Box to ')) {
+                    $('.message-title').text('IoTBox perform a self flashing it take a lot of time (~30min).');
+                    var loadingBlock = $('.loading-block');
+                    var messageStatus = $('.message-status');
+                    var messageError = $('.error-message');
+                    loadingBlock.removeClass('o_hide');
+
+                    messageStatus.text('Prepare space for IoTBox.');
+                    try {
```
Comment (aprieels) [inline addons/hw_posbox_homepage/views/upgrade_page.html]: Can't you have a single try-catch block around the whole process?

**Comment by pimodoo** [comment]
@rhe-odoo could you review this

File: addons/hw_posbox_homepage/controllers/main.py [hunk 2]
```diff
@@ -280,3 +289,71 @@ def clear_six_payment_terminal(self):
...
+    def check_version(self):
+        return helpers.get_version()
+
+    @http.route('/hw_proxy/perform_flashing_create_partition', type='http', auth='none')
+    def perform_flashing_create_partition(self):
+        try:
+            response = subprocess.check_output(['sudo', 'bash', '-c', '. /home/pi/odoo/addons/point_of_sale/tools/posbox/configuration/upgrade.sh; create_partition']).decode().split('\n')[-2]
+            if response == 'Error_Card_Size':
+                raise Exception(response)
+            return 'SUCCESS'
```
Comment (rhe-odoo) [inline addons/hw_posbox_homepage/controllers/main.py]: Wouldn't it be better to make something like: return response('success', status=200)

**Comment by qle-odoo** [comment]
@robodoo r+ rebase-ff

**Pull request closed without merging**